src/main/java/com/johanekstroem/parking/Controller/ParkingEventController.java

```diff
@@ -1,100 +1,66 @@
-import com.johanekstroem.parking.Service.ValidateStopTime;
+
 
 @AllArgsConstructor
 @Controller
 @RequestMapping("/api")
 public class ParkingEventController {
 
 
-  CarRepository carRepository;
-  ParkingSpotRepository parkingSpotRepository;
-  ParkingEventRepository parkingEventRepository;
-  ValidateStopTime validateStopTime;
-  
+ParkingEventService parkingEventService;
 
-  @PostMapping("/parkingevent")
-  public Object startParking(@ModelAttribute("parkingevent") ParkingEvent parkingEvent) {
-    var stoptime = parkingEvent.getStoptime();
-    var saveParkingEvent = parkingEventRepository.save(parkingEvent);
-
-    if (validateStopTime.isInFuture(stoptime)) {
-      Long carID = parkingEvent.getCar().getId();
-      var carOptional = carRepository.findById(carID);
-      if (carOptional.isPresent()) {
-        Car car = carOptional.get();
-        car.addParkingEvent(parkingEvent);
-        carRepository.save(car);
-      }
-
-      Long parkingSpotID = parkingEvent.getParkingSpot().getId();
-      var parkingSpotOptional = parkingSpotRepository.findById(parkingSpotID);
-      if (parkingSpotOptional.isPresent()) {
-        ParkingSpot parkingSpot = parkingSpotOptional.get();
-        parkingSpot.addParkingEvent(parkingEvent);
-        parkingSpotRepository.save(parkingSpot);
-      
-         URI location = ServletUriComponentsBuilder
-                .fromCurrentRequest()
-                .buildAndExpand(saveParkingEvent.getId())
-                .toUri();
-                return "redirect:/saved";
-              }
-    }
 
-    return "redirect:/ops";
-  }
-  
   @GetMapping("/parkingevent")
-  public Iterable<ParkingEvent> getAllParkingEvents() {
-    return parkingEventRepository.findAll();
+  @ResponseBody
+  public List<ParkingEvent> getAllParkingEvents() {
+    return parkingEventService.getAllParkingEvents();
   }
 
-  @GetMapping("/parkingevent/{id}")
-  public ResponseEntity<ParkingEvent> getAllParkingEventsByID(@PathVariable("id") Long id) {
-    var parking = parkingEventRepository.findById(id);
-    return parking.map(parkingEvent -> ResponseEntity.ok().body(parkingEvent)).orElseGet(() -> ResponseEntity.notFound().build());
+  @GetMapping("/parkingevent/{ParkingEventId}")
+  @ResponseBody
+  public ResponseEntity<ParkingEvent> getAllParkingEventsByID(@PathVariable Long ParkingEventId) {
+    var result= parkingEventService.getOneParkingEventByID(ParkingEventId);
+    return result.map(parkingEvent -> ResponseEntity.ok().body(parkingEvent)).orElseGet(() -> ResponseEntity.notFound().build());
   }
 
-   @GetMapping(path = "/parkingevent", params = "filter")
-    public List<ParkingEvent> filterActiveParkings(@RequestParam String filter) {
-        return parkingEventRepository.filterOnActiveParkingEvents();
+  @PostMapping("/parkingevent")
+  public Object startParking(@ModelAttribute("parkingevent") ParkingEvent parkingEvent) {
+    if(parkingEventService.startParking(parkingEvent)){
+      return "redirect:/saved";
```

Comment:
we wouldn't need this part anymore because we implemented React. I know that you wrote these codes before we implemented React anyway! I think we can make a new issue for this :)